cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -191,134 +120,12 @@ private static DataprocClient getDataprocClient(DataprocConf conf, boolean requi
       // If the CDAP instance is running on a GCE/GKE VM from a project that matches the provisioner project,
       // use the network of that VM.
       network = systemNetwork;
-    } else if (network == null) {
-      // Otherwise, pick a network from the configured project using the Compute API
-
-      network = findNetwork(networkHostProjectId, compute);
-    }
-    if (network == null) {
-      throw new IllegalArgumentException("Unable to automatically detect a network, please explicitly set a network.");
     }
 
-    String subnet = conf.getSubnet();
-    Network networkInfo = getNetworkInfo(networkHostProjectId, network, compute);
+    NetworkClient networkClient = requireNetwork ? new ComputeNetworkClient(conf) : new NoOpNetworkClient();
+    networkClient.init(network);
 
-    List<String> subnets = networkInfo.getSubnetworks();
-    if (subnet != null && !subnetExists(subnets, subnet)) {
-      throw new IllegalArgumentException(String.format("Subnet '%s' does not exist in network '%s' in project '%s'. "
-                                                         + "Please use a different subnet.",
-                                                       subnet, network, networkHostProjectId));
-    }
-
-    // if the network uses custom subnets, a subnet must be provided to the dataproc api
-    boolean autoCreateSubnet = networkInfo.getAutoCreateSubnetworks() == null ?
-      false : networkInfo.getAutoCreateSubnetworks();
-    if (!autoCreateSubnet) {
-      // if the network uses custom subnets but none exist, error out
-      if (subnets == null || subnets.isEmpty()) {
-        throw new IllegalArgumentException(String.format("Network '%s' in project '%s' does not contain any subnets. "
-                                                           + "Please create a subnet or use a different network.",
-                                                         network, networkHostProjectId));
-      }
-    }
-
-    subnet = chooseSubnet(network, subnets, subnet, conf.getRegion());
-
-    return new DataprocClient(new DataprocConf(conf, network, subnet), client, compute, networkInfo);
-  }
-
-  private static PeeringState getPeeringState(String systemProjectId, String systemNetwork, Network networkInfo) {
```

Comment:
everything else removed from this class was moved to ComputeNetworkClient as-is.